File: api/internal/domain/team.go
package domain

import (
	"time"
)

type Team struct {
	ID   string `json:"id"`
	Name string `json:"name"`

	XP      int `json:"xp"`
	Balance int `json:"balance"`

	Emoji string `json:"emoji"`
	Color string `json:"color"`

	IsRunner bool `json:"is_runner"`

	VetoPeriodEnd time.Time `json:"veto_period_end"`

	GameID    string    `json:"game_id"`
	CreatedAt time.Time `json:"created_at"`
}

type TeamMember struct {
	TeamID string `json:"team_id"`
	UserID string `json:"user_id"`
}

func (t *Team) IsVeto() bool {
	return time.Now().Before(t.VetoPeriodEnd)
}

type TeamCreate struct {
	Name       string `json:"name"`
	Emoji      string `json:"emoji"`
	Color      string `json:"color"`
	GameInvite string `json:"game_invite"`
	GameID     string `json:"-"`
}

type TeamUpdate struct {
	Name          *string    `json:"name"`
	Emoji         *string    `json:"emoji"`
	Color         *string    `json:"color"`
	Balance       *int       `json:"-"`
	XP            *int       `json:"-"`
	VetoPeriodEnd *time.Time `json:"-"`
}

File: ios/Inertia/Inertia/Models/Team.swift
//
//  Team.swift
//  Inertia
//
//  Created by peony on 31/03/2024.
//

import Foundation

struct Team: Codable {
    var id: String
    var name: String
    var xp: Int
    var balance: Int
    var emoji: String
    var color: String
    var isRunner: Bool
    var vetoPeriodEnd: String
    var gameId: String
    var createdAt: String
    
    enum CodingKeys: String, CodingKey {
        case id = "id"
        case name = "name"
        case xp = "xp"
        case balance = "balance"
        case emoji = "emoji"
        case color = "color"
        case isRunner = "is_runner"
        case vetoPeriodEnd = "veto_period_end"
        case gameId = "game_id"
        case createdAt = "created_at"
    }
    
    static func mock(id: String = "123") -> Team {
        Team(id: id, name: "Penguincat Inc.", xp: 30, balance: 500, emoji: "🐳", color: "#42adf5", isRunner: false, vetoPeriodEnd: "2024-04-30T14:57:17.002268Z", gameId: "123", createdAt: "2024-03-31T14:57:17.002268Z")
    }
}
